src/features/Applet/Schedule/Calendar/YearView/MonthCalendar/CalendarDate/CalendarDate.tsx

```diff
@@ -0,0 +1,119 @@
+import { MouseEvent, useState, useRef } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import theme from 'styles/theme';
+import { StyledBodySmall } from 'styles/styledComponents/Typography';
+import { StyledHeadline } from 'styles/styledComponents/Typography';
+
+import { MonthEvent } from '../../../MonthEvent';
+import { formatToYearMonthDate } from '../../../Calendar.utils';
+import { CalendarDateProps } from './CalendarDate.types';
+import {
+  StyledDayBtn,
+  StyledDotsWrapper,
+  StyledEventDot,
+  StyledTooltipDate,
+  StyledTooltip,
+  StyledTooltipEventWrapper,
+  StyledMonthName,
+  StyledMore,
+} from './CalendarDate.styles';
+import { getDayName } from './CalendarDate.utils';
+
+export const CalendarDate = ({ dateToRender, dateOfMonth, onClick, events }: CalendarDateProps) => {
+  const { t } = useTranslation();
+  const dayBtnRef = useRef<HTMLButtonElement | null>(null);
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const [tooltipPosition, setTooltipPosition] = useState<'bottom' | 'top' | null>(null);
+
+  const open = Boolean(anchorEl);
+  const isToday = formatToYearMonthDate(dateToRender) === formatToYearMonthDate(new Date());
+  const isOffRange =
+    dateToRender.getMonth() < dateOfMonth.getMonth() ||
+    dateToRender.getMonth() > dateOfMonth.getMonth();
+  const showMoreText = events.length > 10;
+  const isTooltipBtm = tooltipPosition === 'bottom';
+
+  const handleTooltipOpen = (event: MouseEvent<HTMLButtonElement>) => {
+    if (dayBtnRef.current) {
+      setTooltipPosition(
+        window.innerHeight - dayBtnRef.current.offsetTop - dayBtnRef.current.offsetHeight < 410
+          ? 'top'
+          : 'bottom',
+      );
+    }
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleTooltipClose = () => setAnchorEl(null);
+
+  const getEventsRows = () =>
+    events.map(
+      (event, index) =>
+        ((showMoreText && index < 9) || !showMoreText) && (
+          <StyledTooltipEventWrapper key={index} bgColor={event.backgroundColor}>
```

Comment:
better to use lodash uniqid instead of key index

Reply:
done